server/src/controllers/hostingRequests.ts

```diff
@@ -0,0 +1,355 @@
+    .execute()
+    .then(items => {
+      return response.status(200).json(alterKeys(items, "hostingRequests"))
+    })
+    .catch(error => {
+      logger.error(error)
+      return response.status(500).json({
+        status: "failure",
+        message: error.message
+      })
+    });
+})
+
+//all h requests
+hostingRequestRouter.get('/', async (request: Request, response: Response) => {
+    const connection = await getConnection();
+    const hRequestRepository = connection.getRepository(HostingRequest)
+
+    const hRequests = await hRequestRepository.find();
+
+    return response.json(alterKeys(hRequests, "hostingRequests"))
+})
+
+// PZD-10
+// zatwierdz badz odrzuc prowadzacego prowadzacego
+// localhost:8000/hrequests/resolve/blabla
+hostingRequestRouter.put('/resolve/:id', async (request: Request, response: Response) => {
+  const id = parseInt(request.params.id)
+  const status = request.body.object.status
+
+  if(!validateValues(status, Status, response)) return
+
+  const connection = await getConnection();
+  const hRequestRepository = await connection.getRepository(HostingRequest)
+
+  await getConnection()
+    .createQueryBuilder()
+    .update(HostingRequest)
+    .set({status})
+    .where("id = :id", { id })
+    .execute()
+    .then(async () => {
+      if(status === "accepted") {
+        const hr = await hRequestRepository.findOne({id})
+
+        if(hr === undefined) {
+          return response.status(500).json({
+            status: "failure",
+            message: "hosting request not found"
+          })
+        }
+
+        const host = hr.user
+        await getConnection()
+        .createQueryBuilder()
+        .update(Class)
+        .set({host})
+        .where("groupKey = :groupKey", { groupKey: hr.class.groupKey })
+        .execute()
+        .then(() => {
+          return response.status(200).json({
+            status: "success"
+          })
+        })
+        .catch(error => {
+          logger.error(error)
+          return response.status(500).json({
+            status: "failure",
+            message: error.message
+          })
+        });
+      }
+    })
+    .catch(error => {
+      logger.error(error)
+      return response.status(500).json({
+        status: "failure",
+        message: error.message
+      })
+    });
+})
+
+// PZD-10
+// zapiszMnieNaKursy(objects: Classes[]): void
+// localhost:8000/hrequests
+hostingRequestRouter.post('/plan', async (request: Request, response: Response) => {
+  const objects = request.body.objects
+  const username = request.body.username
+
+  const connection = await getConnection();
+  const classRepository = connection.getRepository(Class)
+  const userRepository = connection.getRepository(User)
+
+  let newhostingRequests = []
+  for (let groupKey of objects.map((c: { groupKey: string; }) => c.groupKey)) {
+    const clas = await classRepository.findOne({groupKey});
+    const user = await userRepository.findOne({username});
```

Comment:
do not fetch user here. ```username``` doesn't change, so you're fetching the same object multiple times. Get rid of it and do not mess imperative and declarative code. when you move fetching user out of this loop, it may look like this:
```typescript
const newHostingRequests = objects.map(c => c.groupKey)
    .map(async (groupKey) => classRepository.findOne({groupKey}))
    .filter(classEntity => !!classEntity)
    .map(classEntity => ({
        status: "pending",
        user,
        class: classEntity
    });
```